xds/internal/balancer/weightedtarget/weightedtarget.go

```diff
@@ -115,6 +115,16 @@ func (w *weightedTargetBalancer) UpdateClientConnState(s balancer.ClientConnStat
 			w.bg.Add(name, balancer.Get(newT.ChildPolicy.Name))
 			// Not trigger a state/picker update. Wait for the new sub-balancer
 			// to send its updates.
+		} else if newT.ChildPolicy.Name != oldT.ChildPolicy.Name {
```

Comment:
What if the child policy name hasn't changed, but the config has changed? Is that possible? Do we need to handle that?

Reply:
That's handled later. We will forward the config and addresses to the children, `w.bg.UpdateClientConnState()`